distributed/utils.py

```diff
@@ -128,6 +128,11 @@ def get_ipv6(host='2001:4860:4860::8888', port=80):
     return _get_ip(host, port, family=socket.AF_INET6, default='::1')
 
 
+def get_ip_interface(ifname):
+    import psutil
+    return psutil.net_if_addrs()[ifname][0].address
```

Comment:
psutil doesn't specify in which order the results are returned, so you should really select the result for which the `family` member is `AF_INET` (or error out if there is None).